lib/std/dwarf.zig

```diff
@@ -478,9 +478,15 @@ const LineNumberProgram = struct {
             if (file_entry.dir_index >= self.include_dirs.len) return badDwarf();
             const dir_name = self.include_dirs[file_entry.dir_index].path;
 
-            const file_name = try fs.path.join(allocator, &[_][]const u8{
-                dir_name, file_entry.path,
-            });
+            // This will break when the system running this code has different
+            // path separators than what generated the debug info. This was deemed
+            // acceptable as this code is really only designed to be used in
+            // combination with std.debug
+            const parts = if (file_entry.dir_index > 0 and !fs.path.isAbsolute(dir_name))
```

Comment:
I wasn't sure if it would be better stylistically to create two separate arrays and choose between them, or to just create one and pass a different slice of it to `path.join` depending on what behavior we want. I chose this because I think it's a bit more readable, but I'd be fine with the other option as well.